Content.Shared/Wieldable/WieldableSystem.cs

```diff
@@ -94,7 +94,8 @@ private void OnGunWielded(EntityUid uid, GunWieldBonusComponent component, ref I
 
     private void OnDeselectWieldable(EntityUid uid, WieldableComponent component, HandDeselectedEvent args)
     {
-        if (!component.Wielded)
+        if (!component.Wielded ||
+            TryComp(args.User, out HandsComponent? hands) && hands.Hands.Count > 2)
```

Comment:
use HandSystem.EnumerateHands or add a version that returns the count

Reply:
Now using handsystem